commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/MeanTest.java

```diff
@@ -0,0 +1,229 @@
+                Double.MAX_VALUE),
+            Arguments.of(new double[] {-Double.MAX_VALUE, Double.POSITIVE_INFINITY},
+                Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {-Double.MAX_VALUE, -Double.MAX_VALUE},
+                -Double.MAX_VALUE),
+            Arguments.of(new double[] {Double.NEGATIVE_INFINITY, -Double.MIN_VALUE},
+                Double.NEGATIVE_INFINITY),
+            Arguments.of(new double[] {Double.NaN, 34.56, 89.74}, Double.NaN),
+            Arguments.of(new double[] {34.56, Double.NaN, 89.74}, Double.NaN),
+            Arguments.of(new double[] {34.56, 89.74, Double.NaN}, Double.NaN),
+            Arguments.of(new double[] {Double.NaN, 3.14, Double.NaN, Double.NaN},
+                Double.NaN),
+            Arguments.of(new double[] {Double.NaN, Double.NaN, Double.NaN}, Double.NaN),
+            Arguments.of(new double[] {Double.NEGATIVE_INFINITY, Double.MAX_VALUE},
+                Double.NEGATIVE_INFINITY),
+            Arguments.of(new double[] {Double.MAX_VALUE, 1}, Double.MAX_VALUE / 2),
+            Arguments.of(new double[] {-Double.MAX_VALUE, 1, 1}, -Double.MAX_VALUE / 3),
+            Arguments.of(new double[] {-Double.MAX_VALUE, -1, 1}, -Double.MAX_VALUE / 3),
+            Arguments.of(new double[] {Double.MAX_VALUE, -1}, Double.MAX_VALUE / 2),
+            Arguments.of(new double[] {Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY,
+                Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY}, Double.POSITIVE_INFINITY)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testMean")
+    void testParallelStream(double[] values, double expected) {
+        double ans = Arrays.stream(values)
+                .parallel()
+                .collect(Mean::create, Mean::accept, Mean::combine)
+                .getAsDouble();
+
+        TestHelper.assertEquals(expected, ans, 5, () -> "parallel stream");
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testMean")
+    void testMeanRandomOrder(double[] values, double expected) {
+        UniformRandomProvider rng = TestHelper.createRNG();
+        for (int i = 1; i <= 10; i++) {
+            testMean(TestHelper.shuffle(rng, values), expected);
+            testParallelStream(TestHelper.shuffle(rng, values), expected);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = {"testCombine"})
+    void testCombine(double[][] values, double expected) {
+        Mean mean1 = Mean.create();
+        Mean mean2 = Mean.create();
+        Arrays.stream(values[0]).forEach(mean1);
+        Arrays.stream(values[1]).forEach(mean2);
+        double mean2BeforeCombine = mean2.getAsDouble();
+        mean1.combine(mean2);
+        TestHelper.assertEquals(expected, mean1.getAsDouble(), 10, () -> "combine");
+        Assertions.assertEquals(mean2BeforeCombine, mean2.getAsDouble());
+    }
+
+    static Stream<Arguments> testCombine() {
```

Comment:
I think this case is missing where you can get overflow in the combine when computing the deviation from the mean:
```java
            Arguments.of(new double[][] {{Double.MAX_VALUE}, {-Double.MAX_VALUE}}, 0),
            Arguments.of(new double[][] {{1}, {-Double.MAX_VALUE}}, -Double.MAX_VALUE / 2),
            Arguments.of(new double[][] {{1, 1, 1}, {-Double.MAX_VALUE}}, -Double.MAX_VALUE / 4),
```